### PR TITLE
Make the mod server friendly

### PR DESCRIPTION
**Pull request by JustRed23**
The mod was not serverside friendly, crashing on boot, I fixed that.

Some things to note:
- if you do not have a @SubscribeEvent in your class, do not register the class as it will error
- @ OnlyIn(Dist.CLIENT) will actually remove the method that is annotated with that from the mod on the server dist, making those methods completely unknown to Java, causing a NoSuchMethodError

### CODE REVIEW

**Comment by JustRed23** [comment]
Fixes #27 & https://github.com/GregTechCEu/GregTech-Modern/issues/1553

**Comment by JustRed23** [comment]
Apparently the registries were getting mixed between client and serverside as well, as they were not correctly set to null, creating a decoderexception on an attempted login

**Comment by JustRed23** [comment]
Here's an example of what that looked like: https://mclo.gs/FiUYHLF

**Comment by Ashley123477** [comment]
Will verify by testing new commits later on today.

**Comment by Ashley123477** [comment]
After testing this now works and allows you to use Latest GT Modern Versions for 1.21 with servers.

File: src/main/java/com/lowdragmc/lowdraglib/LDLib.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public static boolean isClient() {
 
     public static boolean isRemote() {
         if (isClient()) {
-            return Minecraft.getInstance().isSameThread();
+            return !Minecraft.getInstance().isSameThread();
```
Comment (screret) [inline src/main/java/com/lowdragmc/lowdraglib/LDLib.java:81]: isRemote is the same as `Level#isClientSide`, so this is wrong.

**Review by screret — CHANGES_REQUESTED** [review]

**Comment by screret** [comment]
eh whatever, I'll just fix it myself.

**Comment by JustRed23** [comment]
> eh whatever, I'll just fix it myself.

In that case, change Platform.getFrozenRegistry to use !isRemote instead, as it somehow did not work without me making that change

**Comment by screret** [comment]
I'm fairly certain that it should be using isRemote == true to check for the client side?

**Comment by Ashley123477** [comment]
As of now with those PR commits, it allows the server to launch and the Client to connect, having an issue, crashes out once opening inventory
[crash-2024-07-08_17.43.10-client.txt
[latest.log](https://github.com/user-attachments/files/16132161/latest.log)
](https://github.com/user-attachments/files/16132160/crash-2024-07-08_17.43.10-client.txt)

**Comment by screret** [comment]
yes yes fixing that

**Comment by Ashley123477** [comment]
> yes yes fixing that

Throw the jar over once built and ill test it

**Pull request merged**